src/OpenTelemetry/Internal/InterlockedHelper.cs

```diff
@@ -0,0 +1,58 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+using System.Runtime.CompilerServices;
+
+namespace OpenTelemetry.Internal;
+
+internal static class InterlockedHelper
+{
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public static void Add(ref double location, double value)
+    {
+        // Note: Not calling InterlockedHelper.Read here on purpose because it
+        // is too expensive for fast/happy-path. If the first attempt fails
+        // we'll end up in an Interlocked.CompareExchange loop anyway.
+        double currentValue = Volatile.Read(ref location);
+
+        double updatedValue;
+        unchecked
+        {
+            updatedValue = currentValue + value;
+        }
```

Comment:
The checked/unchecked distinction isn't relevant for double. Even if you were in a checked context, using unchecked here is a nop. You can see at [SharpLab](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBLANgHwAEAmARgFgAoQgZgAIS6BhOgbyrs4foBMIBXYLhh0AsqQAUfQcLo9SaOQKEiexAJR0AvAD46/AHZgAFjDABrGDymk6AajkaA3By60lMkaOJTls+YrSKo6aunQmZpbW8vYhLpQAvkA) that there's no difference in the generated IL whether checked/unchecked is used.

Reply:
Thanks! Updated.